### PR TITLE
Fix sdist, and test Cabal in Travis.

### PR DESCRIPTION
**Pull request by judah**

### CODE REVIEW

File: travis-cabal.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+do
+  echo "Unregistering $p"
+  ghc-pkg unregister --force $p || true
+done
+
+for p in $PACKAGES
+do
+    echo "Cabal building $p"
+    (cd $p &&
+        date
```
Comment (blackgnezdo) [inline travis-cabal.sh]: Do you want to keep the rash of dates?

File: travis-cabal.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        cabal install --enable-tests --only-dependencies
+        date
+        cabal configure --enable-tests
+        date
+        cabal build
+        date
+        cabal sdist
+        date
+        SRC_TGZ=$(cabal info . | awk '{print $2 ".tar.gz";exit}')
```
Comment (blackgnezdo) [inline travis-cabal.sh]: Space before exit?

File: travis-cabal.sh (hunk 1 above)
Comment (judah) [inline travis-cabal.sh]: Done.

File: travis-cabal.sh (hunk 2 above)
Comment (judah) [inline travis-cabal.sh]: Done.

**Pull request merged**